### ISSUE TITLE
5.1.1 User Guide section "Publishing" uses deprecated "classifier" attribute

### ISSUE THREAD

**Issue by gekonwi** [gradle/gradle#8257]
The code sample in [5.1.1 User Guide section "Publishing"](https://docs.gradle.org/current/userguide/publishing_overview.html), sub-section [Adding custom artifacts to a publication](https://docs.gradle.org/current/userguide/publishing_overview.html#sec:publishing_custom_artifacts_to_maven), uses the [deprecated Jar task attribute "classifier"](https://docs.gradle.org/current/dsl/org.gradle.api.tasks.bundling.Jar.html#org.gradle.api.tasks.bundling.Jar:classifier).

### Expected Behavior
The code sample should use the currently recommended, non-deprecated way of adding custom artifacts to a publication. 

### Current Behavior
The code sample uses the [deprecated Jar task attribute "classifier"](https://docs.gradle.org/current/dsl/org.gradle.api.tasks.bundling.Jar.html#org.gradle.api.tasks.bundling.Jar:classifier):

```groovy
task sourcesJar(type: Jar) {
    classifier = 'sources'
    from sourceSets.main.allJava
}

task javadocJar(type: Jar) {
    classifier = 'javadoc'
    from javadoc.destinationDir
}

publishing {
    publications {
        mavenJava(MavenPublication) {
            from components.java

            artifact sourcesJar
            artifact javadocJar
        }
    }
}
```

**Comment by FerusGrim** [comment]
I've recently upgraded to 5.1.1 and started getting deprecation warnings regarding this issue.

This issue is, notably, the only reference I've been able to find to this deprecation and I cannot, for the life of me, find the _proper_ non-deprecated way to handle this.

I realize that it's outside of the scope of your issue, but do you think you could elaborate on the updated method? I'd appreciate it. <3

**Comment by big-guy** [comment]
Thanks for noticing this. This is fixed on release and will be in 5.2

**Comment by langsmith** [comment]
what pull request resolved this issue?
